src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs

```diff
@@ -779,27 +789,41 @@ protected override void Dispose(bool disposing)
                     // Dispose the message stream (note that we OWN this stream, so we always dispose it).
                     this.messageOutputStream.Dispose();
                 }
+
+                if (this.binaryValueStream != null)
+                {
+                    this.binaryValueStream.Flush();
+                    this.binaryValueStream.Dispose();
+                }
+
+                if (this.stringWriter != null)
+                {
+                    this.stringWriter.Flush();
+                    this.stringWriter.Dispose();
+                }
             }
             finally
             {
                 this.messageOutputStream = null;
                 this.asynchronousOutputStream = null;
+                this.binaryValueStream = null;
                 this.textWriter = null;
                 this.jsonWriter = null;
+                this.stringWriter = null;
             }
 
             base.Dispose(disposing);
         }
 
-        private static IJsonWriter CreateJsonWriter(IServiceProvider container, TextWriter textWriter, bool isIeee754Compatible)
+        private static IJsonWriter CreateJsonWriter(IServiceProvider container, Stream stream, TextWriter textWriter, bool isIeee754Compatible)

```

Comment:
>Stream stream, [](start = 80, length = 14)

where does stream use?

Reply:
Good catch; this was from an earlier design. Fixed.

---
In reply to: [259080473](https://github.com/OData/odata.net/pull/1204#discussion_r259080473) [](ancestors = 259080473)